labs/src/main/scala/org/scalalabs/basic/lab01/OOExercise.scala

```diff
@@ -40,6 +40,32 @@ import scala.language.implicitConversions
  *   of type [[org.scalalabs.basic.lab01.CurrencyConverter]]
  * - Use the implicit CurrencyConverter to do the conversion. 
  */
-class Euro {
 
+abstract class Currency(val symbol:String)
+
+class Euro(val euro:Int, val cents:Int=0) extends Currency("EUR") with Ordered[Euro]{
+  lazy val inCents = 100 * euro + cents
+  
+  def +(addMe:Euro) = {
+    val centsSum = inCents + addMe.inCents
+    Euro.fromCents(centsSum)
+ }
+  
+  def *(factor:Int) = {
+    val centsProduct = inCents * factor
+    Euro.fromCents(centsProduct)
+  }
+  
+  override def toString = {
+    if (cents == 0)
+      f"$symbol: $euro,--"
+    else
+      f"$symbol: $euro,$cents%02d"
```

Comment:
Nice. I did this ever so slightly differently and I think your way is more elegant.